### PR TITLE
[exporter/logzioexporter] Added scope name to exported logs

### PR DESCRIPTION
**Pull request by tamir-michaeli**
**Description:** <Describe what has changed.>
In cases when an instrumentation library is used to export logs, the scope field is not being copied to the exported logs.
With this patch, if the scope field is populated in the original log, the scopeName field in the exported log will contains the name of the scope. 
Ex. Adding a feature - Explain what this achieves.-->

**Link to tracking Issue:** #20659 

**Testing:** Added a test with a scope name.
Changed previous tests to send an empty scope name.

**Documentation:** Added a note in the readme.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -59,7 +59,7 @@ exporter/jaegerthrifthttpexporter/                       @open-telemetry/collect
 exporter/kafkaexporter/                                  @open-telemetry/collector-contrib-approvers @pavolloffay @MovieStoreGuy
 exporter/loadbalancingexporter/                          @open-telemetry/collector-contrib-approvers @jpkrohling
 exporter/logicmonitorexporter/                           @open-telemetry/collector-contrib-approvers @bogdandrutu @khyatigandhi6 @avadhut123pisal
-exporter/logzioexporter/                                 @open-telemetry/collector-contrib-approvers @jkowall @Doron-Bargo @yotamloe
+exporter/logzioexporter/                                 @open-telemetry/collector-contrib-approvers @mirii1994 @Doron-Bargo @yotamloe
```
Comment (atoulme) [inline .github/CODEOWNERS:62]: Please make this a separate PR. Is @mirii1994 a member of the OpenTelemetry organization?

Comment (tamir-michaeli) [reply .github/CODEOWNERS:62]: She is not, will send her the instructions, and will split the codeowner changes to a seperate PR.

**Comment by wideberg** [comment]
Hey, just checking to see if there's any progress on this?

**Comment by tamir-michaeli** [comment]
@atoulme Mind taking another look?

**Comment by yotamloe** [comment]
Hey @atoulme :) Are there any more changes needed for this PR to be reviewed?

File: exporter/logzioexporter/README.md [hunk 2]
```diff
@@ -169,3 +169,183 @@ service:
     logs:
       level: debug #activate debug mode
 ```
+
+#### Scope Name
+When using the logs exporter with logs originating from instrumentation library (i.e opentelemetry log4j2 appender), the scopeName field will be added (if the field is populated in the original log). 
+
+[beta]:https://github.com/open-telemetry/opentelemetry-collector#beta
+[contrib]:https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
+# Logzio Exporter
```
Comment (atoulme) [inline exporter/logzioexporter/README.md:178]: All this part can go, this is now generated at the top of the file.

File: exporter/logzioexporter/README.md [hunk 3]
```diff
@@ -169,3 +169,183 @@ service:
...
+  telemetry:
+    logs:
+      level: debug #activate debug mode
+```
+
+#### Scope Name
+When using the logs exporter with logs originating from instrumentation library (i.e opentelemetry log4j2 appender), the scopeName field will be added (if the field is populated in the original log).
+No additional configuration needed.
+
+[beta]:https://github.com/open-telemetry/opentelemetry-collector#beta
```
Comment (atoulme) [inline exporter/logzioexporter/README.md:350]: Remove

File: exporter/logzioexporter/README.md (hunk 2 above)
Comment (tamir-michaeli) [reply exporter/logzioexporter/README.md]: Done

File: exporter/logzioexporter/README.md (hunk 3 above)
Comment (tamir-michaeli) [reply exporter/logzioexporter/README.md]: Done

**Comment by codeboten** [comment]
Pinging code owners for feedback on this one @jkowall @Doron-Bargo @yotamloe

**Comment by jkowall** [comment]
I'm going to remove myself from codeowners, I am not involved anymore with this codebase. Thanks.

File: exporter/logzioexporter/README.md [hunk 4]
```diff
@@ -169,3 +169,173 @@ service:
     logs:
       level: debug #activate debug mode
 ```
+
+#### Scope Name
+When using the logs exporter with logs originating from instrumentation library (i.e opentelemetry log4j2 appender), the scopeName field will be added (if the field is populated in the original log). 
+
+
+| Status                   |                       |
```
Comment (atoulme) [inline exporter/logzioexporter/README.md:177]: Looks like starting from this all this text is a duplicate?

File: exporter/logzioexporter/exporter_test.go [hunk 5]
```diff
@@ -64,6 +62,7 @@ func fillLogOne(log plog.LogRecord) {
 	attNestedMap := attMap.PutEmptyMap("nested")
 	attNestedMap.PutStr("string", "v1")
 	attNestedMap.PutDouble("number", 499)
+
```
Comment (atoulme) [inline exporter/logzioexporter/exporter_test.go:65]:
```suggestion
```

File: exporter/logzioexporter/exporter_test.go [hunk 6]
```diff
@@ -92,6 +91,21 @@ func fillLogNoTimestamp(log plog.LogRecord) {
...
+	log.SetTimestamp(TestLogTimestamp)
+	log.SetDroppedAttributesCount(2)
+	log.SetSeverityNumber(plog.SeverityNumberInfo)
+	log.SetSeverityText("Info")
+	log.SetSpanID([8]byte{0x01, 0x02, 0x04, 0x08})
+	log.SetTraceID([16]byte{0x08, 0x04, 0x02, 0x01})
+	attrs := log.Attributes()
+	attrs.PutStr("app", "log4j2")
+	attrs.PutDouble("instance_num", 1)
+	attrs.PutStr("scopeName", "test.class")
```
Comment (atoulme) [inline exporter/logzioexporter/exporter_test.go:104]: That’s not where scope name usually goes. Look at scope records, the parent of the log record.

File: exporter/logzioexporter/jsonlog_test.go [hunk 7]
```diff
@@ -66,15 +73,30 @@ func TestConvertLogRecordToJSON(t *testing.T) {
...
 		},
 	}
 	for _, test := range convertLogRecordToJSONTests {
-		output := convertLogRecordToJSON(test.log, test.resource)
+		output := convertLogRecordToJSON(test.log, test.scopeName, test.resource)
 		require.Equal(t, output, test.expected)
 	}
 
 }
+
```
Comment (atoulme) [inline exporter/logzioexporter/jsonlog_test.go:99]:
```suggestion
```

File: exporter/logzioexporter/exporter_test.go [hunk 8]
```diff
@@ -51,11 +51,9 @@ func fillLogOne(log plog.LogRecord) {
 	log.SetSeverityText("Info")
 	log.SetSpanID([8]byte{0x01, 0x02, 0x04, 0x08})
 	log.SetTraceID([16]byte{0x08, 0x04, 0x02, 0x01})
-
```
Comment (atoulme) [inline exporter/logzioexporter/exporter_test.go:54]: Bring back the deleted pines to reduce the changeset

File: .chloggen/logzio-log-scope.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: add scopename to exported logs
+
+# One or more tracking issues related to the change
+issues: [20659]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: when exists, scope name will be add to exported logs under the scopeName field.
```
Comment (atoulme) [inline .chloggen/logzio-log-scope.yaml:16]:
```suggestion
subtext: when it exists, scope name will be added to exported logs under the scopeName field.
```

**Comment by codeboten** [comment]
Pinging code owners for review here @Doron-Bargo @yotamloe

File: exporter/logzioexporter/README.md (hunk 4 above)
Comment (codeboten) [reply exporter/logzioexporter/README.md:177]: @tamir-michaeli was this added by accident?

Comment (tamir-michaeli) [reply exporter/logzioexporter/README.md]: @codeboten indeed, removed with latest commit.

File: exporter/logzioexporter/exporter_test.go (hunk 6 above)
Comment (tamir-michaeli) [reply exporter/logzioexporter/exporter_test.go:102]: @atoulme @codeboten Care to provide a more detailed explanation? :)

Comment (MovieStoreGuy) [reply exporter/logzioexporter/exporter_test.go:102]: There is a type named `ScopeLogRecords` which contains the "scope" attributes and should exist in that type.

You can see where you have used it in the `exporter` file :)

**Review by MovieStoreGuy — CHANGES_REQUESTED** [review]
I believe you can make this a whole bunch simpler here by merging attributes maps together which should help make this easier to extend in future if needed.

**Comment by tamir-michaeli** [comment]
@codeboten Might taking another look?

**Review by yotamloe — APPROVED** [review]
Nice job! approved

**Review by codeboten — COMMENTED** [review]
Thanks for the review @yotamloe, and for your patience @tamir-michaeli.

@MovieStoreGuy you have a request for change, can you take another look

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by atoulme** [comment]
@MovieStoreGuy mind merging too?
